presto-raptorx/src/main/java/com/facebook/presto/raptorx/metadata/CommitCleaner.java

```diff
@@ -175,6 +195,41 @@ public synchronized void removeOldCommits()
         }
     }
 
+    // TODO(taozhao) For workers, shardDao doesn't have to be a list, each one only deals with its own node_id. May write a separate Cleaner later.
+    public void removeOldWorkerTransactions(long nodeId)
+    {
+        ShardCommitCleanerDao thisNodeDao = shardDao.get(dbShard(nodeId, shardDao.size()));
+
+        Set<Long> transactionIds = thisNodeDao.getSuccessfulTransactionIds(nodeId);
+        // May need to shard CreatedChunks by nodeId too; but in the near future there is only one Mysql, so can live with it for some time.
+        for (Iterable<Long> ids : partition(transactionIds, 1000)) {
+            for (ShardCommitCleanerDao shard : shardDao) {
+                shard.cleanupCreatedChunks(ids);
+            }
+            thisNodeDao.cleanupTransactions(ids);
+        }
+
+        // transactionManager here contains only this node self's transactions
```

Comment:
Instead of bookkeeping everything in comment, we should restructure the code to separate `TransactionManager/NodeManager` for workers and coordinator. 

At the very least, like your previous comment said, we should have separate commit cleaner.

Reply:
TransactionManager cannot separate. The variable and function are all used in both. NodeManager, maybe, will do it another task